discovery/gossiper.go

```diff
@@ -1092,6 +1092,7 @@ func (d *AuthenticatedGossiper) networkHandler() {
 							"barrier shutdown: %v",
 							err)
 					}
+					announcement.err <- err
```

Comment:
Select on quit here.

Reply:
I don't think that is necessary, as the channel is buffered? We don't do that any other places.